### ISSUE TITLE
else body not recognized in while loop

### ISSUE THREAD

**Issue by Crestwave** [elves/elvish#820]
According to the [docs](https://elv.sh/ref/language.html#conditional-loop:-while), this should execute if the condition is initially false, but it results in a compilation error.

```elvish
~> while $false { echo a } else { echo b }
Compilation error: too many arguments
[tty], line 1: while $false { echo a } else { echo b }
```

**Comment by 0x005c** [comment]
This issue is solved by #863. Should be closed.

**Comment by Crestwave** [comment]
@0x005c You can mention one of `[close{,s,d}, fix{,es,ed}, resolve{,s,d}]`, then the issue number in the pull request description to automatically close the issue when merged. E.g., "resolves #863".
